src/core_modules/capture-core/components/WidgetAddEvent/WidgetAddEvent.component.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+import React, { type ComponentType } from 'react';
+import i18n from '@dhis2/d2-i18n';
+import { EnrollmentRegistrationEntry, SingleEventRegistrationEntry }
+    from 'capture-core/components/DataEntries';
+import { spacersNum } from '@dhis2/ui';
+import withStyles from '@material-ui/core/styles/withStyles';
+import { scopeTypes } from '../../metaData';
+import { useScopeInfo } from '../../hooks/useScopeInfo';
+import { useRegistrationFormInfoForSelectedScope }
+    from '../DataEntries/common/useRegistrationFormInfoForSelectedScope';
+import { NonBundledDhis2Icon } from '../NonBundledDhis2Icon';
+import { Widget } from '../Widget';
+import type { Props } from './WidgetAddEvent.types';
+
+const styles = () => ({
+    header: {
+        display: 'flex',
+        alignItems: 'center',
+        padding: spacersNum.dp8,
+    },
+    icon: {
+        paddingRight: spacersNum.dp8,
+    },
+});
+
+const WidgetAddEventPlain = ({ programStage, currentScopeId, resultsPageSize, classes }: Props) => {
+    const { icon, stageForm } = programStage;
+    const { scopeType } = useScopeInfo(currentScopeId);
+    const { registrationMetaData } = useRegistrationFormInfoForSelectedScope(currentScopeId);
+
+    const onSaveRegistrationEntry = () => {};
+    const renderDuplicatesDialogActions = () => <></>;
+    const renderDuplicatesCardActions = () => <></>;
+
+    return (
+        <Widget
+            noncollapsible
+            header={
+                <div className={classes.header}>
+                    {icon && (
+                        <div className={classes.icon}>
+                            <NonBundledDhis2Icon
+                                name={icon?.name}
+                                color={icon?.color}
+                                width={30}
+                                height={30}
+                                cornerRadius={2}
+                            />
+                        </div>
+                    )}
+                    <span> {stageForm.name} </span>
+                </div>
+            }
+
+        >
+            <div data-test="add-event-form">
+                {scopeType === scopeTypes.TRACKER_PROGRAM && <EnrollmentRegistrationEntry
```

Comment:
Hey, 
I am not sure `EnrollmentRegistrationEntry` is the component to use here. `EnrollmentRegistrationEntry` is used to register a new **enrollment** and this feature is about a new **event**. 
I think you should use `SingleEventRegistrationEntry` and `NewEventDataEntry` (or something similar from the single event page) and make them work for an enrollment's event (tracker program).  
In the Cypress tests, you should check for the **event** inputs fields just like you did a few comments back https://github.com/dhis2/capture-app/pull/1908/commits/6b44aee44b09c0398638b842613e28305a6691b4 . 
You can check how this feature works in the old Tracker capture app.  

<img width="2293" alt="Screenshot 2021-07-20 at 16 01 34" src="https://user-images.githubusercontent.com/6817073/126337706-d8bdf57c-b426-45c4-adb1-fc7846b93a73.png">

<img width="1701" alt="Screenshot 2021-07-20 at 16 03 26" src="https://user-images.githubusercontent.com/6817073/126337799-f0f30fca-6761-4777-bd9c-0f48b973109e.png">

Let me know if this makes sense. Thank you! 😄

Reply:
sorry I was confused by the fact that there isn't possible to create any new event on the main page, only enrolment as you said. I will make the `SingleEventRegistrationEntry` usable for both Tracker and Event program.